packages/presentation/lib/star_movie_app.dart

```diff
@@ -18,50 +22,69 @@ class StarMovieApp extends StatefulWidget {
 class _StarMovieAppState extends BlocScreenState<StatefulWidget, AppBloc> {
   @override
   Widget build(BuildContext context) {
-    return MaterialApp(
-      debugShowCheckedModeBanner: false,
-      title: 'Star Movie app',
-      theme: ThemeData(
-        primarySwatch: Colors.red,
-      ),
-      themeMode: ThemeMode.dark,
-      darkTheme: ThemeData(
-        brightness: Brightness.dark,
-        primaryColorDark: AppColorsDark.primaryColorDark,
-        canvasColor: AppColorsDark.canvasColor,
-      ),
-      localizationsDelegates: const [
-        SM.delegate,
-        GlobalMaterialLocalizations.delegate,
-        GlobalWidgetsLocalizations.delegate,
-        GlobalCupertinoLocalizations.delegate,
-      ],
-      supportedLocales: SM.delegate.supportedLocales,
-      home: StreamBuilder<BlocData>(
-        stream: bloc.dataStream,
-        builder: (context, result) {
-          final blocData = result.data;
-          final appData = blocData?.data;
-          if (appData is AppData && blocData != null) {
-            return Scaffold(
-              body: Navigator(
-                onPopPage: (Route<dynamic> route, dynamic result) {
-                  bloc.handleRemoveRouteSettings(route.settings);
-                  return route.didPop(result);
-                },
-                pages: appData.pages.toList(),
-              ),
-              bottomNavigationBar: appData.isButtonNavBarActive
-                  ? AppBottomNavigationBar(
-                      bloc: bloc,
-                blocData: appData,
-                    )
-                  : const SizedBox(),
-            );
-          }
-          return Container();
-        },
-      ),
+    return ScreenUtilInit(
+      designSize: Dimens.figmaScreenSize,
+      useInheritedMediaQuery: true,
+      builder: (context, child) {
+        return MaterialApp(
+          debugShowCheckedModeBanner: false,
+          title: 'Star Movie app',
+          theme: ThemeData(
+            primarySwatch: Colors.red,
+          ),
+          themeMode: ThemeMode.dark,
+          darkTheme: ThemeData(
+            brightness: Brightness.dark,
+            primaryColorDark: AppColorsDark.primaryColorDark,
+            canvasColor: AppColorsDark.canvasColor,
+          ),
+          localizationsDelegates: const [
+            SM.delegate,
+            GlobalMaterialLocalizations.delegate,
+            GlobalWidgetsLocalizations.delegate,
+            GlobalCupertinoLocalizations.delegate,
+          ],
+          supportedLocales: SM.delegate.supportedLocales,
+          home: StreamBuilder<BlocData>(
+            stream: bloc.dataStream,
+            builder: (context, result) {
+              final blocData = result.data;
+              final appData = blocData?.data;
+              if (appData is AppData && blocData != null) {
+                return Scaffold(
```

Comment:
return its own scaffold with content for desktop and mobile. don't keep whole logic in one

Reply:
to avoid the big class with lot of check for different platforms